### PR TITLE
[feat] Add QR Code Scanning Functionality

### PR DESCRIPTION
**Pull request by hummusonrails**
Add the ability to scan a recipient wallet address for a new NFT minting via QR code

### CODE REVIEW

File: package-lock.json
Comment (VadimSaveljev) [inline package-lock.json:1]: This shouldn't be pushed, because yarn is used for packages

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "react-dom": "18.2.0",
     "react-dropzone": "14.2.3",
     "react-router-dom": "6.14.2",
+    "react-zxing": "^1.1.4",
```
Comment (VadimSaveljev) [inline package.json]: need exact version, like other dependencies
yarn add react-zxing@1.1.4 -E

File: src/components/Pages/MyAssets/MintNft/MintNft.tsx [hunk 2]
```diff
@@ -2,24 +2,20 @@ import { BN } from '@polkadot/util';
 import { ChangeEvent, FormEvent, memo, useCallback, useEffect, useRef, useState } from 'react';
 import { FormControl, Stack } from 'react-bootstrap';
 import { Link, useParams } from 'react-router-dom';
+import { useZxing } from "react-zxing";
 
 import { saveImageToIpfs } from '@api/pinata.ts';
-
```
Comment (VadimSaveljev) [inline src/components/Pages/MyAssets/MintNft/MintNft.tsx:7]: Please leave these new lines,
yarn format does the job automatically for this part
but the ones in HTML below were done manually by me for clarity, I like to separate blocks visually, so leave them be

**Review by VadimSaveljev — COMMENTED** [review]
I'll check the solution itself a bit later, having calls today
but for now these are the changes I'd like to see

**Review by VadimSaveljev — CHANGES_REQUESTED** [review]
So some changes need to be done:
1. react-zxing is requiring specific node versions, 16 or 18, I'm using 20 right now, so I need a "hack" for this `yarn install --ignore-engines`, so either the lib itself should be updated to not have this limit or some other lib should be used instead.
2. UX improvements: 
a. A QR code icon can be placed next to the "Mint to" input or label
b. On click of that icon, a modal should appear (use modal from react-bootstrap, like other modals in the portal)
c. Said modal should only have the QR scanner in it and nothing else
d. On successful scan modal should close automatically and the wallet address should appear in the input

**Comment by hummusonrails** [comment]
> So some changes need to be done...

Thanks, I'll work on incorporating those changes

**Comment by hummusonrails** [comment]
@SaltyCucumber, could use your help when you have a moment.

I refactored the QR code scanning integration to incorporate all the requested UX improvements. I needed to shift to a new package that worked better in a modal than the original one I chose. 

However, I'm experiencing a `tsc` linting error for the `new` constructor of the `QrScanner`, even though the `QrScanner` types definition in `node_modules` seems to define it.

Specifically, seeing the following error:

```
src/components/Modals/QrScanner/QrScanner.tsx:36:27 - error TS2351: This expression is not constructable.
  Type 'typeof import("/Dev/nft-portal/node_modules/qr-scanner/types/qr-scanner")' has no construct signatures.

36     const qrScanner = new QrScanner(
                             ~~~~~~~~
Found 1 error in src/components/Modals/QrScanner/QrScanner.tsx:36
```

I've tried to explicitly add the types file to the config, but that doesn't seem to work.

Besides this issue, everything from a usability perspective is working as requested.

**Comment by VadimSaveljev** [comment]
Unfortunately the lib confuses typescript a bit, so I used "any" in this case. And refactored the code.

**Review by VadimSaveljev — APPROVED** [review]

**Pull request merged**